website/src/docs/companion.md

```diff
@@ -285,38 +285,38 @@ const options = {
   filePath: 'path/to/download/folder',
   sendSelfEndpoint: 'localhost:3020',
   secret: 'mysecret',
-  uploadUrls: ['https://myuploadurl.com', 'http://myuploadurl2.com'],
+  uploadUrls: ['https://myuploadurl.com', /^http:\/\/myuploadurl2.com\//],
   debug: true,
   metrics: false,
 }
 ```
 
 1. **filePath(required)** - Full path to the directory to which provider files would be downloaded temporarily.
 
-2. **redisUrl(optional)** - URL to running Redis server. If this is set, the state of uploads would be stored temporarily. This helps for resumed uploads after a browser crash from the client. The stored upload would be sent back to the client on reconnection.
+2. **secret(required)** - A secret string which Companion uses to generate authorization tokens.
 
-3. **redisOptions(optional)** - An object of [options supported by redis client](https://www.npmjs.com/package/redis#options-object-properties). This option can be used in place of `redisUrl`.
+3. **uploadUrls(required)** - An allowlist (array) of strings (exact URLs) or regular expressions. If specified, Companion will only accept uploads to these URLs. This is needed to make sure a Companion instance is only allowed to upload to your servers. **Omitting this leaves your system open to potential [SSRF](https://en.wikipedia.org/wiki/Server-side_request_forgery) attacks.** Example: []
```

Comment:
It says it's `required` but there is a warning implying you can actually omit it... Which is true?

Reply:
This is an intentional lie. I want to encourage developers to specify `uploadUrls` so I write it as **required** even though it is not really required. There was a wish for this change to be backward-compatible and not having to release a new major of companion. #3180

Reply:
Should we use a different word for that? Maybe we can also warn that it will be required in a future semver-major.
```suggestion
3. **uploadUrls(recommended)** - An allowlist (array) of strings (exact URLs) or regular expressions. If specified, Companion will only accept uploads to these URLs. This is needed to make sure a Companion instance is only allowed to upload to your servers. **Omitting this leaves your system open to potential [SSRF](https://en.wikipedia.org/wiki/Server-side_request_forgery) attacks, and may throw an error in futures `@uppy/companion` releases.** Example: []
```

Reply:
I was thinking about writing "recommended" too, but I wrote "required" because IMO it's so important that the user specify this, because if it's not specified then it can be a severe security risk.
Maybe @Murderlon or @kvz has some opinions.

If we change it to "recommended" then we should also change `secret` to "recommended" for consistency.

Reply:
I think the warning in the text is severe enough that we can accept Antoine's proposal